ckcp/hack/util/destroy.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Pipeline Service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SCRIPT_DIR="$(
+
+  cd "$(dirname "$0")" >/dev/null || exit
+  pwd
+)"
+
+CONFIG="$SCRIPT_DIR/../../config.yaml"
+
+KUBECONFIG=${KUBECONFIG:-$HOME/.kube/config}
+
+# Removes different Argo CD applications deployed
+uninstall_pipeline_service(){
+    printf "\n Uninstalling Pipeline Service:\n"
+    # Remove Pipeline-Service application
+    kubectl delete -f "$SCRIPT_DIR/../../../gitops/argocd/argocd.yaml"
+    # Remove all the applications deployed by Pipeline-Service
+    kubectl delete -k "$SCRIPT_DIR/../../../gitops/argocd/argo-apps"
+}
+
+# Uninstalls operators
+uninstall_operators(){
+    # Openshift-Pipelines Operator
+    printf "\n Uninstalling Openshift-Pipelines Operator:\n"
+    kubectl delete tektonconfigs.operator.tekton.dev config
+    kubectl delete subscriptions.operators.coreos.com -n openshift-operators openshift-pipelines-operator
+    kubectl delete csv openshift-pipelines-operator-rh.v1.7.2 -n openshift-operators
+
+
+    # cert-manager
+    printf "\n Uninstalling cert-manager Operator:\n"
+    kubectl delete -k "$SCRIPT_DIR/../../../operators/cert-manager"
+
+
+    # GitOps
+    printf "\n Uninstalling Openshift-GitOps Operator:\n"
+    kubectl delete -k "$SCRIPT_DIR/../../openshift-operators/openshift-gitops"
+    kubectl delete csv openshift-gitops-operator.v1.5.6 -n openshift-operators
+
+    # PAC
+    printf "\n Uninstalling PAC controllers:\n"
+    kubectl delete -k "$SCRIPT_DIR/../../../gitops/pac/manifests"
+}
+
+# Cleans ckcp deployed resources from the cluster
+uninstall_ckcp(){
+    # remove resoures created by ckcp
+    printf "\n Uninstalling ckcp from the cluster:\n"
+    kubectl delete -k "$SCRIPT_DIR/../../openshift/overlays/dev"
+
+    # remove syncer resources
+    printf "\n Removing resources taken up by the syncer:\n"
+    current_context="$(yq e ".current-context" <"$KUBECONFIG")"
+    current_context=$(echo "$current_context" | sed  's,default/,,g; s,:6443/kube:admin,,g')
+    kubectl delete -f /tmp/syncer-"$current_context".yaml
+
+    # remove namesapces generated by ckcp
+    printf "\n Removing namespaces created by ckcp:\n"
+    ckcp_generated_n="$(kubectl get ns | cut -d " " -f 1 | grep kcp)"
```

Comment:
this command fetches all namespaces with substring 'kcp'. Would need to be changed to fetch ns starting with 'kcp-' as the generated namespaces follow this pattern.

```
[bnr@bnr pipelines-service]$ k get ns | cut -d " " -f1  | grep kcp
ckcp
kcp-1tfpg81ue6h0
kcp-mujo6gr3n42i
kcp-syncer-0d30625518f0617f253a43fc9339ea91-ziyfk3sn
kcp-zbbksthmbir1
```

Reply:
I've changed this part by first fetching a label which would be applied to all the namespaces created by the kcp-ws in the compute. And then selecting the NS by the label selector.